query/stdlib/influxdata/influxdb/provider.go

```diff
@@ -0,0 +1,166 @@
+package influxdb
+
+import (
+	"context"
+	"fmt"
+
+	arrowmemory "github.com/apache/arrow/go/arrow/memory"
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/ast"
+	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/dependencies/influxdb"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/memory"
+	"github.com/influxdata/flux/values"
+	"github.com/influxdata/influxdb/v2/kit/platform"
+	"github.com/influxdata/influxdb/v2/kit/platform/errors"
+	"github.com/influxdata/influxdb/v2/query"
+	"github.com/influxdata/influxdb/v2/storage/reads/datatypes"
+)
+
+type (
+	Config       = influxdb.Config
+	Predicate    = influxdb.Predicate
+	PredicateSet = influxdb.PredicateSet
+)
+
+// Provider is an implementation of influxdb.Provider that exposes the
+// query.StorageReader to flux and, if a host or org were specified, it
+// delegates to the influxdb.HttpProvider.
+type Provider struct {
+	influxdb.HttpProvider
+	Reader       query.StorageReader
+	BucketLookup BucketLookup
+}
+
+func (p Provider) SeriesCardinalityReaderFor(ctx context.Context, conf influxdb.Config, bounds flux.Bounds, predicateSet influxdb.PredicateSet) (influxdb.Reader, error) {
+	// If an organization is specified, it must be retrieved through the http
+	// provider.
+	if conf.Org.IsValid() || conf.Host != "" {
```

Comment:
I get why `conf.Host != ""` requires going through HTTP, but am less sure about the `conf.Org.IsValid()`. Is it a security thing?

Reply:
As it currently works, if an org is specified, the query will use the HTTP provider to get the information, essentially proxying the request with the provided org/host/token info. This is mostly relevant for cloud, but I kept that capability here so that you could hypothetically run an identical query against OSS as you would run on cloud and have it behave the same. Typically I would expect an OSS query to not specify an org/host/token.

The actual organization for auth is picked up from the context via the query API. I'm thinking it might be useful down the road to allow for specifying an org in the flux query and have it use that org on that server.